### ISSUE TITLE
Can't format with codespell

### ISSUE THREAD

**Issue by eeeXun** [jose-elias-alvarez/null-ls.nvim#1279]
### FAQ

- [X] I have checked the [FAQ](https://github.com/jose-elias-alvarez/null-ls.nvim#faq) and it didn't resolve my problem.

### Issues

- [X] I have checked [existing issues](https://github.com/jose-elias-alvarez/null-ls.nvim/issues?q=is%3Aissue) and there are no issues with the same problem.

### Neovim Version

v0.8.1

### Dev Version?

- [X] I am using a stable Neovim release version, or if I am using a dev version of Neovim I have confirmed that my issue is reproducible on a stable version.

### Operating System

Archlinux

### Minimal Config

```Lua
-- this template is borrowed from nvim-lspconfig
local on_windows = vim.loop.os_uname().version:match("Windows")

local function join_paths(...)
	local path_sep = on_windows and "\\" or "/"
	local result = table.concat({ ... }, path_sep)
	return result
end

vim.g.loaded_remote_plugins = ""
vim.cmd([[set runtimepath=$VIMRUNTIME]])

local temp_dir = vim.loop.os_getenv("TEMP") or "/tmp"

vim.cmd("set packpath=" .. join_paths(temp_dir, "nvim", "site"))

local package_root = join_paths(temp_dir, "nvim", "site", "pack")
local install_path = join_paths(package_root, "packer", "start", "packer.nvim")
local compile_path = join_paths(install_path, "plugin", "packer_compiled.lua")

local null_ls_config = function()
	local null_ls = require("null-ls")
	local formatting = null_ls.builtins.formatting
	local diagnostics = null_ls.builtins.diagnostics
	-- add only what you need to reproduce your issue
	null_ls.setup({
		sources = {
			diagnostics.codespell,
			formatting.codespell,
		},
		debug = true,
	})
end

local function load_plugins()
	-- only add other plugins if they are necessary to reproduce the issue
	require("packer").startup({
		{
			"wbthomason/packer.nvim",
			{
				"jose-elias-alvarez/null-ls.nvim",
				requires = { "nvim-lua/plenary.nvim" },
				config = null_ls_config,
			},
		},
		config = {
			package_root = package_root,
			compile_path = compile_path,
		},
	})
end

if vim.fn.isdirectory(install_path) == 0 then
	vim.fn.system({ "git", "clone", "https://github.com/wbthomason/packer.nvim", install_path })
	load_plugins()
	require("packer").sync()
else
	load_plugins()
	require("packer").sync()
end
```

### Steps to Reproduce

Run: `nvim --clean -u minimal_init.lua`
Insert text `alha`
Run: `:lua vim.lsp.buf.format()`

### Reproducibility Check

- [X] I confirm that my minimal config is based on the `minimal_init.lua` template and that my issue is reproducible by running `nvim --clean -u minimal_init.lua` and following the steps above.

### Expected Behavior

`alha` formatted to `alpha`

### Actual Behavior

`alha` not formatted

### Debug Log

[TRACE Tue 06 Dec 2022 09:49:50 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:123: received LSP request for method initialize
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:148: received LSP notification for method initialized
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:148: received LSP notification for method textDocument/didOpen
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] ...ack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_DIAGNOSTICS_ON_OPEN
[DEBUG Tue 06 Dec 2022 09:49:50 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:320: spawning command "codespell" at /tmp with args { "-" }
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:204: error output: 1: alha
	alha ==> alpha

[TRACE Tue 06 Dec 2022 09:49:50 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:205: output: nil
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] ...ck/packer/start/null-ls.nvim/lua/null-ls/diagnostics.lua:181: received diagnostics from source 1
[TRACE Tue 06 Dec 2022 09:49:50 PM CST] ...ck/packer/start/null-ls.nvim/lua/null-ls/diagnostics.lua:182: { {
    col = 0,
    end_col = 4,
    end_lnum = 0,
    lnum = 0,
    message = "alha ==> alpha",
    row = 1,
    severity = 2,
    source = "codespell"
  } }
[TRACE Tue 06 Dec 2022 09:49:58 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:123: received LSP request for method textDocument/formatting
[TRACE Tue 06 Dec 2022 09:49:58 PM CST] ...ack/packer/start/null-ls.nvim/lua/null-ls/generators.lua:21: running generators for method NULL_LS_FORMATTING
[DEBUG Tue 06 Dec 2022 09:49:58 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:320: spawning command "codespell" at /tmp with args { "--write-changes", "/tmp/.null-ls_814785_a.txt" }
[TRACE Tue 06 Dec 2022 09:49:58 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:204: error output: nil
[TRACE Tue 06 Dec 2022 09:49:58 PM CST] ...t/null-ls.nvim/lua/null-ls/helpers/generator_factory.lua:205: output: nil
[TRACE Tue 06 Dec 2022 09:50:10 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:123: received LSP request for method shutdown
[TRACE Tue 06 Dec 2022 09:50:10 PM CST] .../site/pack/packer/start/null-ls.nvim/lua/null-ls/rpc.lua:148: received LSP notification for method exit

### Help

No

### Requirements

- [X] I have read and followed the instructions above and understand that my issue will be closed if I did not provide the required information.

**Comment by eeeXun** [comment]
After seeing the log, I can't find `/tmp/.null-ls_814785_a.txt`. Is this expected?

**Comment by jose-elias-alvarez** [comment]
> After seeing the log, I can't find `/tmp/.null-ls_814785_a.txt`. Is this expected?

Yes, this is expected, since the temporary file should be cleaned up after each run. I played around with the `codespell` CLI directly and found that it does nothing when a filename starts with `.` The following patch theoretically fixes the issue on our end, though I think this is ultimately an upstream issue:

```diff
diff --git a/lua/null-ls/loop.lua b/lua/null-ls/loop.lua
index 867aec3..040a62b 100644
--- a/lua/null-ls/loop.lua
+++ b/lua/null-ls/loop.lua
@@ -229,7 +229,7 @@ M.temp_file = function(content, bufname, dirname)
     dirname = dirname or vim.fn.fnamemodify(bufname, ":h")
     local base_name = vim.fn.fnamemodify(bufname, ":t")
 
-    local filename = string.format(".null-ls_%d_%s", math.random(100000, 999999), base_name)
+    local filename = string.format("null-ls_%d_%s", math.random(100000, 999999), base_name)
     local temp_path = u.path.join(dirname, filename)
 
     local fd, err = uv.fs_open(temp_path, "w", 384)
```

I wasn't able to find a matching upstream issue, but it's pretty trivial to replicate. I'd prefer to seek a solution there first if you want to open an issue.

```sh
echo "alha" > .bad.txt && codespell --write-changes .bad.txt # no output, doesn't write to file
echo "alha" > bad.txt && codespell --write-changes bad.txt   # outputs FIXED: bad.txt and writes to file
```
